ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md

```diff
@@ -0,0 +1,108 @@
+
+#### Jinja2 Templates Guidelines
+
+1. **Code Indentation:** Keep less indented code to improve readability.
+2. **Variable Naming:** Use meaningful variable names.
+3. **Use AVD filters:** Use AVD filters for code optimization - [AVD Filters](https://avd.arista.com/5.0/docs/plugins/Filter_plugins/add_md_toc.html).
+4. **Natural Sorting:** Use `arista.avd.natural_sort` for sorting the `for loops` after checking on EOS CLI.
+5. **Defined Checks:**
+    - Avoid `arista.avd.defined` check for parent keys when directly checking for child keys.
+    - Avoid `arista.avd.defined` check for primary and required keys.
+    - Avoid`arista.avd.defined` check when using `arista.avd.default()` and `arista.avd.natural_sort` filters.
+6. **Password Security:** Avoid displaying passwords in the documentation template and use the `arista.avd.hide_passwords` filter to hide it.
+7. **Config Order:** Ensure the order and indentation of configuration matches EOS CLI.
+8. **Exclamation Marks:** Place exclamation marks `!` correctly as per the EOS running-config.
+9. Along with EOS config template, update the documentation template as well (if required).
+10. Implement only commands visible in running-config of the EOS device.
+11. Validate the template using j2lint tool, run `pre-commit run j2lint --all`.
+
+### Build Schemas and Documentation
+
+Run `pre-commit run schemas --all` to re-generate the eos_cli_config_gen schema with any modifications. This command should be executed every time the schema is changed, even if only a description is updated.
+It also updates the documentation with new options.
+
+### Add Molecule Tests
+
+Add some molecule tests in the `ansible_collections/arista/avd/molecule/eos_cli_config_gen` scenario exercising the new knob configuration.
+
+When marking any key as "deprecated," move the related tests to the `eos_cli_config_gen_deprecated_vars` molecule scenario and add any missing tests if necessary.
+
+### Run Molecule Tests
+
+Run `molecule converge -s eos_cli_config_gen` from the path `ansible_collections/arista/avd/` to execute the molecule tests locally and generate the new expected configuration and documentation for newly added test-cases.
+
+Check the PyAVD test coverage report by running `tox -e coverage,report` and work on improving the coverage where possible.
+
+### Update Documentation
+
+If the proposed feature requires any changes to the documentation, make sure to update it accordingly.
+If there are any breaking changes introduced, document them in the porting guide.
+
+### Run Pre-commit Checks
+
+Run all pre-commit checks `pre-commit run --all` to ensure that all files added or modified are correctly following the coding standards and formatting rules.
```

Comment:
```suggestion
Run all pre-commit checks `pre-commit run` to ensure that all files added or modified are correctly following the coding standards and formatting rules.
```

Reply:
`pre-commit run` checks staged files only

Reply:
right. But pre-commit run --all takes forever :)

Reply:
I see it still reads `--all` is it expected? (conversation was resolved)

Reply:
Since it was resolved, I didn't update it, assuming it's fine to keep it as it is.